data/trie/patriciaMerkleTrie.go

```diff
@@ -682,16 +664,24 @@ func (tr *patriciaMerkleTrie) GetOldRoot() []byte {
 	return tr.oldRoot
 }
 
-// Close calls all the canceFuncs
+// Close stops all the active goroutines started by the trie
 func (tr *patriciaMerkleTrie) Close() error {
 	tr.mutOperation.Lock()
 	defer tr.mutOperation.Unlock()
 
-	for _, cancelFunc := range tr.cancelFuncs {
-		cancelFunc()
+	if !isChannelClosed(tr.chanClose) {
+		close(tr.chanClose)
 	}
 
-	tr.cancelFuncs = make(map[string]context.CancelFunc)
-
 	return nil
 }
+
+func isChannelClosed(ch chan struct{}) bool {
```

Comment:
this works in conjunction with the mutex, otherwise it can still panic. Will create a component that we should use in all our closing components. :+1: